### PR TITLE
feat(ENG-44122): Add support for AWS Lambda

### PR DESCRIPTION
**Pull request by Crateros**

### CODE REVIEW

**Review by NamelessHH — APPROVED** [review]

File: cmd/httpserver/main.go [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+	conf, err := config.LoadConfiguration()
+	if err != nil {
+		panic(err)
+	}
+
+	// Validate the configuration for HTTP server
+	if err := app.Validator.Validate.Struct(conf.HTTP); err != nil {
+		panic(err)
+	}
+
```
Comment (jasonsites) [inline cmd/httpserver/main.go]: This file really doesn't need to be updated at all. The `config` gets loaded by the `resolver`, no matter which way the app is run.

File: cmd/lambda/main.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+func main() {
+	conf, err := config.LoadConfiguration()
+	if err != nil {
+		panic(err)
+	}
+
+	// Validate the configuration for Lambda
+	if err := app.Validator.Validate.Struct(conf.Lambda); err != nil {
+		panic(err)
+	}
```
Comment (jasonsites) [inline cmd/lambda/main.go]: Same here (circumventing part of the resolver's functionality.)

File: config/config.go [hunk 3]
```diff
@@ -50,6 +51,10 @@ type HTTP struct {
 	} `validate:"required"`
 }
 
+type Lambda struct {
+	LogLevel string `validate:"oneof=debug info warn error"`
```
Comment (jasonsites) [inline config/config.go]: Why is there a separate log level here? The Logger is app-wide.

File: internal/resolver/loaders.go [hunk 4]
```diff
@@ -183,6 +186,47 @@ func (r *Resolver) HTTPServer() *httpserver.Server {
 	return r.httpServer
 }
 
+type LambdaEvent struct {
+	EventType string `json:"event_type"`
+}
```
Comment (jasonsites) [inline internal/resolver/loaders.go]: This belongs in the `lambda` package.

File: internal/resolver/loaders.go [hunk 5]
```diff
@@ -183,6 +186,47 @@ func (r *Resolver) HTTPServer() *httpserver.Server {
...
+
+func UnhandledMethod() (events.APIGatewayProxyResponse, error) {
+	var errorMethodNotAllowed = http.StatusText(http.StatusMethodNotAllowed)
+
+	return events.APIGatewayProxyResponse{
+		StatusCode: http.StatusMethodNotAllowed,
+		Body:       errorMethodNotAllowed,
+	}, errors.New(errorMethodNotAllowed)
+}
```
Comment (jasonsites) [inline internal/resolver/loaders.go]: This is your `lambda` package code.

File: cmd/httpserver/main.go (hunk 1 above)
Comment (Crateros) [reply cmd/httpserver/main.go]: @jasonsites We no longer require the HTTP struct so I figured it'd be worthwhile to validate the struct **if** indeed the app intends to bootstrap as HTTP server.

**Review by jasonsites — APPROVED** [review]

**Pull request merged**